File: hessapp-view/src/app/dto/Group.ts
import {Spend} from "./Spend";
import {Participant} from "./Participant";
import {Message} from "./Message";

export class Group {
  groupID: string;
  name: string;
  description: string;
  moderator: string;
  participants: Participant[];
  spends: Spend[];
  messages: Message[];
  unaspiringMessageCount: number;
}

File: login-service/src/main/java/com/hassapp/api/dto/Group.java
package com.hassapp.api.dto;

import com.hassapp.api.client.group.dto.Spend;
import com.hassapp.api.client.message.dto.Message;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Group {
    private String groupID;
    private String name;
    private String description;
    private String moderator;
    private List<Participant> participants;
    private List<Spend> spends;
    private List<Message> messages;
    private int unaspiringMessageCount;
}
